cep-9999.md

```diff
@@ -0,0 +1,55 @@
+# CEP ???? - Build provenance metadata
+
+<table>
+<tr><td> Title </td><td> CEP ???? -  Build provenance metadata </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 10, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 10, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> https://github.com/conda-forge/conda-smithy/pull/1577 </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize how the conda ecosystem leverages free-form recipe metadata
+to annotate provenance of its published artifacts. Two features are applied:
+
+- `extra` section in the recipe metadata
+- `extra` metadata in `about.json` (via CLI flag `--extra-meta`)
+
+## Motivation
+
+Provenance metadata is useful to assess how and when a conda artifact was built.
+
+Since late 2023, thanks to conda-smithy 3.28.0 and later, conda-forge feedstocks have been
+adding CI provenance in the produced artifacts. `defaults` also applies the same conventions.
+This is used by apps like `conda-metadata-app` to show provenance information in the build details. See the table in this [Python 3.13 example](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fpython-3.13.2-hf636f53_101_cp313.conda).
+
+Additional provenance metadata can be collected for source origins, which can
+be useful for efforts like dependency mapping across ecosystems. See
+[PEP 725](https://peps.python.org/pep-0725/) for practical applications in the
+context of PyPI <> conda interoperability. This type of provenance is out of
+scope for this CEP and may be discussed separately.
+
+## Specification
+
+CI pipelines MUST instrument the build workflows to add the following metadata keys
+via `--extra-meta` or equivalent:
+
+- `sha`: Required. Commit hash of the feedstock being built.
+- `remote_url`: Required on CI. Repository URL of the feedstock being built.
+- `flow_run_id`: Optional. CI-specific identifier for the workflow run.
+
+For local workflows such as those specified by CFEP 03, `remote_url` MAY be omitted, but authors strongly recommend providing the adequate value manually if necessary.
```

Comment:
```suggestion
For local workflows such as those specified by [CFEP-03](https://github.com/conda-forge/cfep/blob/main/cfep-03.md), `remote_url` MAY be omitted, but authors strongly recommend providing the adequate value manually if necessary.
```
Also, if `remote_url` is omitted, should `sha` also be omitted?

Reply:
Hm, I had assumed users interested in provenance are already using some type of version control, but maybe we can't force that either.

About the dash in CFEP 03, see https://github.com/conda/ceps/blob/971cb232407d2d8c5b23394bcae876725b2f12e7/.pre-commit-config.yaml#L35.

Reply:
please keep `sha` as mandatory ... some packages may not have the `remote_url` to not expose private repositories to the public, but the `sha` is this helpful for internal audits / attestations to check that e.g. the `sha` actually exists / was reviewed as part of an PR / triggered a CI run etc.

Reply:
I am writing this to describe the current conventions, not to impose on how packages should be built. I think that should be decided by packaging organizations separately. I plan to submit a CFEP for conda-forge where we do "require these fields in CI pipelines, as recommended by CEP XYZ".

Someone using conda-build to share an artifact with their research lab internally may not need to care about whether the recipe is version controlled or what a git hash is.